toolchain/parse/testdata/var/fail_unused.carbon

```diff
@@ -0,0 +1,43 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/parse/testdata/var/fail_unused.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/parse/testdata/var/fail_unused.carbon
+
+var unused x: i32;
+// CHECK:STDERR: fail_unused.carbon:[[@LINE+4]]:5: error: found `ref` inside `var` pattern [RefInsideVar]
+// CHECK:STDERR: var ref unused y: i32;
+// CHECK:STDERR:     ^~~
+// CHECK:STDERR:
+var ref unused y: i32;
+var unused _: i32;
```

Comment:
Is this supposed to have an error?

We want to avoid mixing failures and passes in the same file split, so that if one flips to the other, we get a FAIL from the overall test run.

This set of tests here seems both a little oddly specific but also beyond the scope of `unused` so I am not sure what to make of it.

It's not testing putting `unused` in all possible positions, but it is testing putting `var` and `ref` together. So maybe we need more tests here, to verify parsing unused in all the positions.

I would suggest using function parameters, cuz you can't even write `ref` on a variable decl.